language/move-prover/stackless-bytecode-generator/src/eliminate_imm_refs.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    function_target::{FunctionTarget, FunctionTargetData},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{
+        AssignKind,
+        Bytecode::{self, *},
+        Operation::*,
+    },
+};
+use spec_lang::{env::FunctionEnv, ty::Type};
+
+pub struct EliminateImmRefsProcessor {}
+
+impl EliminateImmRefsProcessor {
+    pub fn new() -> Box<Self> {
+        Box::new(EliminateImmRefsProcessor {})
+    }
+}
+
+impl FunctionTargetProcessor for EliminateImmRefsProcessor {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        func_env: &FunctionEnv<'_>,
+        mut data: FunctionTargetData,
+    ) -> FunctionTargetData {
+        let func_target = FunctionTarget::new(func_env, &data);
+        data.code = data
```

Comment:
This line of code was interesting.  I wanted to take ownership of each individual bytecode so I don't have to clone all of them.  Instead, I can reuse the one that is not modified.  But I couldn't figure out how to do it since func_target holds on an immutable reference to data inside it.

Reply:
One trick here is to "detach" the code from `data` as your FunctionTarget likely doesn't need it. So you do

```
let code = std::mem::replace(&mut data.code, vec![]);
```

Now you own code, and data has an empty code vector. At the end of transformation, you assign `data.code = new_code`.

Reply:
Done.